File: 729-my-calendar-i/my-calendar-i.ts
class MyCalendar {
    private calendar: number[][];

    constructor() {
        this.calendar = []
    }

    book(start: number, end: number): boolean {
        for(let [s, e] of this.calendar) {
            if (s < end && start < e) return false
        }
        this.calendar.push([start, end])
        return true
    }
}

/**
 * Your MyCalendar object will be instantiated and called as such:
 * var obj = new MyCalendar()
 * var param_1 = obj.book(start,end)
 */

File: 729-my-calendar-i/my-calendar-i.py
class MyCalendar:

    def __init__(self):
        self.calendar = []

    def book(self, start: int, end: int) -> bool:
        for [s, e] in self.calendar:
            if s < end and e > start: # o-----o o-----o
                return False
        
        self.calendar.append([start, end])
        return True


# Your MyCalendar object will be instantiated and called as such:
# obj = MyCalendar()
# param_1 = obj.book(start,end)